src/core/gltf2importer/bufferaccessorparser.cpp

```diff
@@ -106,6 +109,74 @@ Qt3DRender::QAttribute::VertexBaseType accessorTypeFromJSON(int componentType)
         return Qt3DRender::QAttribute::Float;
     }
 }
+
+template<class IndexType>
+bool copySparseValues(int count,
+                      const QByteArray &indicesBuffer, int indicesOffset,
+                      const QByteArray &valuesBuffer, int valuesOffset,
+                      QByteArray &targetBuffer, int attributeOffset, int stride,
+                      int valueSize)
+{
+    if (stride == 0)
+        stride = valueSize;
+    const IndexType *indices = reinterpret_cast<const IndexType *>(indicesBuffer.data() + indicesOffset);
+    for (int i = 0; i < count; ++i) {
+        IndexType ndx = indices[i];
+        char *target = targetBuffer.data() + stride * ndx + attributeOffset;
+        const char *source = valuesBuffer.data() + valuesOffset + i * valueSize;
+        memcpy(target, source, static_cast<size_t>(valueSize));
+    }
+    return true;
+}
+
+template<class IndexType>
+bool copySparseValues(int count,
+                      const QByteArray &indicesBuffer, int indicesOffset,
+                      const QByteArray &valuesBuffer, int valuesOffset,
+                      QByteArray &targetBuffer, int attributeOffset, int stride,
+                      Qt3DRender::QAttribute::VertexBaseType componentType, int numComponents)
+{
+    switch (componentType) {
+    case Qt3DRender::QAttribute::UnsignedByte:
+    case Qt3DRender::QAttribute::Byte:
+        return copySparseValues<IndexType>(count,
```

Comment:
do we really need all this switch again? you already have the type needed in the `typename IndexType`, no?

Reply:
that's not the type of the index, it's the type of the data

Reply:
But arent you doing a switch to call this function and then doing another switch to call the other copySparseValues overload? My point is that at this point you dont need the switch because you can take directly sizeof(IndexType)

Reply:
you need 2 switches. The outer one is on the type of the index, the inner (in `copySparseValues`) is on the type of the values that get copied.